pkgs/by-name/libresoc/openpower-isa.nix

```diff
@@ -0,0 +1,135 @@
+        --replace 'powerpc64le-linux-gnu-gcc' '${pkgsCross.powernv.buildPackages.gcc.targetPrefix}gcc'
+      substituteInPlace src/openpower/simulator/envcmds.py \
+        --replace 'powerpc64-linux-gnu-' '${pkgsCross.powernv.buildPackages.gcc.targetPrefix}' \
+        --replace 'os.environ.get(cmd.upper(), default)' 'default'
+    '';
+
+    patches = [
+      # pip will try to clone dependencies using @git versions, but this patch makes it so we can use the repos that have already been vendored in by nix
+      ./use-vendored-git-dependencies.patch
+      # upstream includes .gitignore in empty directory to place that empty directory in the build output
+      # Nix happens to break this fragile hack, so remove and make output directory manually
+      ./remove-gitignore-check.patch
+      # Makefile attempts to use recently-built binaries via $PATH, expose instead with a $(OPENPOWER) prefix
+      # see postInstall notes for complement
+      ./prefixed-openpower-isa-tools.patch
+    ];
+
+    # Native is the build machine architecture (e.g. x86_64 linux)
+    # This will run a python emulator of the target architecture, which is PowerPC for this project
+    # The assembler has to run on native but target PowerPC assembly
+    propagatedNativeBuildInputs = [
+      astor
+      cached-property
+      cffi
+      libresoc-pyelftools
+      mdis
+      nmigen
+      nmutil
+      pkgsCross.powernv.buildPackages.gcc
+      ply
+      pygdbmi
+    ];
+
+    # TODO: potential upstream work
+    postInstall =
+      ''
+        # complement of `remove-gitignore-check.patch`
+        mkdir -p $out/${python39.sitePackages}/openpower/decoder/isa/generated
+
+        # this file is missed in the installation configuration, manually move into output
+        # potential to upstream fix to configuration that maintains this file
+        cp ./src/openpower/simulator/memmap $out/${python39.sitePackages}/openpower/simulator/
+      ''
+      + (
+        # this project's build steps do not map cleanly onto Nix's build stages.
+        # setuptools builds the the `entry_points` into binaries that are used to codegen
+        # directly into the source directory, which is then bundled into a wheel and installed.
+        #
+        # for Nix, there's a chicken-and-egg problem with this:
+        # only once the package has been built can the codegen be run to... build the package from source.
+        # as a result, the installPhase is overloaded to effectively double as the build phase,
+        # so, the majority of build steps actually occur here, at the end:
+        let
+          codegen = writeShellApplication {
+            name = "run-codegen";
+            runtimeInputs = [gnumake pkgsCross.powernv.buildPackages.gcc];
+            text = "make generate";
+          };
+        in ''
+          # copy special-purpose python modules into path expected by codegen
+          cp -rT ./openpower $out/${python39.sitePackages}/../openpower/
```

Comment:
I think this is a bit brittle. Maybe only take `python39` as an input, then, in a `let` binding make sure that you get `pkgs` from that. So...

```
{
  python39,
}:
let
  python = python39;
  pythonPackages = python.pkgs;
in
python.buildPythonPackage {
  postInstall = ''
    yaddayadda ${python.sitePackages}
  '';
  nativeCheckInputs = with pythonPackages [
    ...
  ];
}
```